clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy

```diff
@@ -154,4 +158,30 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
     task.updateStatus BASE_PHASE, "Done deploying load balancers."
     operationResult
   }
+
+  private void updateApplicationSecurityGroupIngressRules(Map<String, String> securityGroupNamesToIds,
+                                                          ArrayList<Listener> listeners,
+                                                          RegionScopedProviderFactory.RegionScopedProvider regionScopedProvider) {
+    Names names = Names.parseName(description.getName())
+    String application = names.getApp()
+    try {
+      if (application in securityGroupNamesToIds) {
+        String applicationSecurityGroup = securityGroupNamesToIds[application]
+        task.updateStatus BASE_PHASE, "Configuring ingress rules for application security group $applicationSecurityGroup ..."
+        List<IpPermission> permissions = listeners.collect {
```

Comment:
Agreed about calling authorizeSecurityGroup if we don't need to. That call will fail with nvalidPermission.Duplicate on trying to add a rule that already exist.